pkg/chunk/cache/fifo_cache.go

```diff
@@ -166,12 +185,23 @@ func (c *FifoCache) Store(ctx context.Context, keys []string, bufs [][]byte) {
 
 // Stop implements Cache.
 func (c *FifoCache) Stop() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	c.entries = make(map[string]*cacheEntry)
+	c.first = nil
+	c.last = nil
+	c.currSizeBytes = 0
+
+	c.entriesEvicted.Add(float64(len(c.entries)))
+	c.entriesCurrent.Set(float64(0))
+	c.memoryBytes.Set(float64(0))
 }
 
 // Put stores the value against the key.
 func (c *FifoCache) Put(ctx context.Context, keys []string, values []interface{}) {
```

Comment:
Given that this cache is only used with byte slices, why don't we modify `Put` method to accept byte slices only, and simplify the size calculation code? It's a lot of code that's not used.

Reply:
Let's address that in a separate PR

Reply:
Actually, I went ahead and changed it in this PR. Thank you for suggesting. The code became more clear.

Reply:
Thanks. This was a different suggestion :) I'd like to see it done, but not necessarily in this PR.